cf-custom-resources/test/trigger-state-machine-test.js

```diff
@@ -0,0 +1,200 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+"use strict";
+
+describe("trigger state machine", () => {
+  const aws = require("aws-sdk-mock");
+  const lambdaTester = require("lambda-tester").noVersionCheck();
+  const nock = require("nock");
+  const sinon = require("sinon");
+  const handler = require("../lib/trigger-state-machine");
+
+  const responseURL = "https://cloudwatch-response-mock.example.com/";
+  const logGroup = "/aws/lambda/testLambda";
+  const logStream = "2021/06/28/[$LATEST]9b93a7dca7344adeb193d15c092dbbfd";
+  const testRequestId = "f4ef1b10-c39a-44e3-99c0-fbf7e53c3943";
+
+  const stateMachineARN = "arn::mock::statemachine";
+  const syncExecProps = {
+    SourceBucket: "mockSrcBucket",
+    AssetMappingFilePath: "path/to/file",
+    DestinationBucket: "mockDstBucket",
+  };
+
+  const origConsole = console;
+
+  handler.withDeadlineExpired(() => {
+    return new Promise((resolve, reject) => { });
+  });
+
+  afterEach(() => {
+    aws.restore();
+  });
+  afterAll(() => {
+    console = origConsole;
+  });
+
+  test("bogus operation fails", () => {
+    console.error = () => { };
+    const request = nock(responseURL)
+      .put("/", (body) => {
+        return (
+          body.Status === "FAILED" &&
+          body.Reason ===
+          "Unsupported request type bogus (Log: /aws/lambda/testLambda/2021/06/28/[$LATEST]9b93a7dca7344adeb193d15c092dbbfd)"
+        );
+      })
+      .reply(200);
+    return lambdaTester(handler.handler)
+      .context({
+        logGroupName: logGroup,
+        logStreamName: logStream,
+      })
+      .event({
+        ResponseURL: responseURL,
+        RequestType: "bogus",
+        RequestId: testRequestId,
+        ResourceProperties: {},
+        LogicalResourceId: "mockID",
+      })
+      .expectResolve(() => {
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  test("delete event is a no-op", () => {
+    const request = nock(responseURL)
+      .put("/", (body) => {
+        return body.Status === "SUCCESS";
+      })
+      .reply(200);
+    return lambdaTester(handler.handler)
+      .context({
+        logGroupName: logGroup,
+        logStreamName: logStream,
+      })
+      .event({
+        ResponseURL: responseURL,
+        RequestType: "Delete",
+        RequestId: testRequestId,
+        ResourceProperties: {},
+        LogicalResourceId: "mockID",
+      })
+      .expectResolve(() => {
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  test("happy path", () => {
+    const request = nock(responseURL)
+      .put("/", (body) => {
+        return body.Status === "SUCCESS";
```

Comment:
we can porlly test that the `PhysicalResourceId` is not changing in these test cases as well!